File: Microsoft .NET/.NET Framework 4/ASP.NET/VB.NET/MVC Web Reporting Sample/MVC Web Reporting Sample/Controllers/LLWebReportDesignerController.vb
Imports combit.Reporting
Imports combit.Reporting.Web.WebReportDesigner.Server

Namespace WebReporting.Controllers
    Public Class LLWebReportDesignerController
        Inherits WebReportDesignerController

        Public Overrides Sub OnProvideListLabel(ByVal provideListLabelContext As ProvideListLabelContext)
            Dim ll As ListLabel = DefaultSettings.GetListLabelInstance(provideListLabelContext.RepositoryId, Nothing)
            provideListLabelContext.NewInstance = ll
        End Sub

        Public Overrides Sub OnProvideRepository(ByVal provideFileRepositoryContext As ProvideRepositoryContext)
            provideFileRepositoryContext.FileRepository = DefaultSettings.GetBaseRepository()
        End Sub

        Public Overrides Sub OnProvideWebReportDesignerSessionOptions(ByVal provideWebReportDesignerSessionOptionsContext As ProvideWebReportDesignerSessionOptionsContext)
            MyBase.OnProvideWebReportDesignerSessionOptions(provideWebReportDesignerSessionOptionsContext)
        End Sub

        Public Overrides Sub OnProvideProhibitedActions(ByVal provideProhibitedActionsContext As ProvideProhibitedActionsContext)
            For Each action In DefaultSettings.GetProhibitedActions()
                provideProhibitedActionsContext.ProhibitedActions.Add(action)
            Next
        End Sub
    End Class
End Namespace

File: Microsoft .NET/.NET 6/ASP.NET/React MVC Web Reporting Sample/Controllers/LLWebReportDesignerController.cs
using System.Security.Principal;
using System.Web;
using combit.Reporting;
using combit.Reporting.Web.WebReportDesigner.Server;
using combit.Reporting.Web.WebReportDesigner.Server.JsonModels;

namespace ReactMVCWebReportingSample.Controllers
{
    public class LLWebReportDesignerController : WebReportDesignerController
    {
        public override void OnProvideListLabel(ProvideListLabelContext provideListLabelContext)
        {
            ListLabel ll = DefaultSettings.GetListLabelInstance(provideListLabelContext.RepositoryId);

            provideListLabelContext.NewInstance = ll;
        }

        public override void OnProvideRepository(ProvideRepositoryContext provideFileRepositoryContext)
        {
            provideFileRepositoryContext.FileRepository = DefaultSettings.GetBaseRepository();
        }

        public override void OnProvideWebReportDesignerSessionOptions(ProvideWebReportDesignerSessionOptionsContext provideWebReportDesignerSessionOptionsContext)
        {
            base.OnProvideWebReportDesignerSessionOptions(provideWebReportDesignerSessionOptionsContext);
        }

        public override void OnProvideProhibitedActions(ProvideProhibitedActionsContext provideProhibitedActionsContext)
        {
            foreach (WebReportDesignerAction action in DefaultSettings.GetProhibitedActions())
            {
                provideProhibitedActionsContext.ProhibitedActions.Add(action);
            }
        }
    }
}
